src/fdb5/tools/fdb-move-mpi.cc

```diff
@@ -0,0 +1,197 @@
+            throw UserError(ss.str(), Here());
+        }
+    } else { // this process is just a data mover - subscribing to receive tasks
+        char task[1024];
+        
+        while (true) {
+            int ready = 1;
+
+            comm.template send<int>(&ready, 1, 0, 0);
+            comm.template receive<char>(task, 1024, 0, 0);
+
+            if (task[0] != '\0') {
+                Log::debug() << "fdb-move mover (rank " << rank_ << ") received task " << task << std::endl;
+
+                FileCopy fileCopy(task, 1024);                
+                fileCopy.execute();
+            } else {
+                Log::debug() << "fdb-move mover (rank " << rank_ << ") done" << std::endl;
+                break;
+            }
+        }
+    }
+}
+
+
+void FDBMove::execute(const CmdArgs& args) {
+
+    if (!mpi_ || rank_ == 0) {
+        FDB fdb(config(args));
+
+        size_t count = 0;
+        for (const FDBToolRequest& request : requests("read")) {
+            if (count) {
+                std::stringstream ss;
+                ss << "Multiple requests are not supported" << std::endl;
+                throw eckit::UserError(ss.str());
+            }
+
+            if (request.all()) {
+                std::stringstream ss;
+                ss << "Move ALL not supported. Please specify a single database." << std::endl;
+                throw eckit::UserError(ss.str(), Here());
+            }
+
+            // check that the request is only referring a single DB - no ranges of values
+            const metkit::mars::MarsRequest& req = request.request();
+            std::vector<std::string> params = req.params();
+            for (const std::string& param: params) {
+                const std::vector<std::string>& values = req.values(param);
+
+                if (values.size() != 1) {
+                    std::stringstream ss;
+                    ss << "Move requires a single value for each parameter in the request." << std::endl << "Parameter " << param << "=" << values << " not supported." << std::endl;
+                    throw eckit::UserError(ss.str(), Here());
+                }
+            }
+
+            // check that exaclty one DB matches
+            StatsIterator it = fdb.stats(request);
+            StatsElement se;
+            if (!it.next(se)) {
+                std::stringstream ss;
+                ss << "Request " << req << " does not matches with an existing database. Please specify a single database." << std::endl;
+                throw eckit::UserError(ss.str(), Here());
+            }
+            if (it.next(se)) {
+                std::stringstream ss;
+                ss << "Request " << req << " matches with more than one existing database. Please specify a single database." << std::endl;
+                throw eckit::UserError(ss.str(), Here());
+            }
+
+            MoveIterator list = fdb.move(request, destination_, !keep_, removeDelay_, mpi_, threads_);
```

Comment:
This looks structurally like a non-MPI version of this could be created fairly straightforwardly from the same source file?